thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java

```diff
@@ -155,48 +196,41 @@ private String generatePresenterId(@NonNull final TiPresenter presenter) {
      * doesn't exist
      */
     @Nullable
-    private synchronized ActivityScopedPresenters getScope(final Activity activity) {
-        final ActivityInstanceObserver detector = mActivityInstanceObserver;
-        if (detector == null) {
-            return null;
-        }
+    private synchronized PresenterScope getScope(final Object host) {
+        if (host instanceof Activity) {
+            final ActivityInstanceObserver detector = mActivityInstanceObserver;
+            if (detector == null) {
+                return null;
+            }
 
-        final String scopeId = detector.getActivityId(activity);
-        if (scopeId == null) {
-            return null;
-        }
+            final Activity activity = (Activity) host;
+            final String scopeId = detector.getActivityId(activity);
+            if (scopeId == null) {
+                return null;
+            }
 
-        return mScopes.get(scopeId);
+            return mScopes.get(scopeId);
+        } else {
+            // currently only Activity is supported as host
+            throw new IllegalStateException(
+                    "Host has unknown type " + host.getClass().getSimpleName());
+        }
     }
 
     /**
-     * Returns an existing scope or creates a new one for the given Activity. Registers a callback
-     * so the scope will be cleaned up when the Activity finishes
+     * registers the {@link ActivityInstanceObserver.ActivityFinishListener} for the activity
      *
-     * @return an existing or new created scope for presenters
+     * @param activity to listen for the finish event
+     * @param hostId   id to track the Activity across orientation changes
      */
-    @NonNull
-    private synchronized ActivityScopedPresenters getScopeOrCreate(final Activity activity) {
-        registerActivityObserver(activity);
-
-        final String scopeId = mActivityInstanceObserver.getActivityId(activity);
-        final ActivityScopedPresenters scope;
-
-        if (scopeId == null) {
-            final String newScopeId = mActivityInstanceObserver.startTracking(activity);
-            scope = new ActivityScopedPresenters();
-            mScopes.put(newScopeId, scope);
+    private void observeActivityFinish(final Activity activity, @Nullable final String hostId) {
+        final ActivityInstanceObserver observer = registerActivityObserver(activity);
+        if (hostId != null) {
+            // stack tracking of this untracked Activity
+            observer.startTracking(activity, hostId);
         } else {
-            final ActivityScopedPresenters savedScope = mScopes.get(scopeId);
-            if (savedScope == null) {
-                scope = new ActivityScopedPresenters();
-                mScopes.put(scopeId, scope);
-            } else {
-                scope = savedScope;
-            }
+            // already observing this activity
```

Comment:
We don't need that else block... 
Is that comment really worth it to keep it? 🤔 
I don't think so...